src/script/lua_api/l_object.cpp

```diff
@@ -1546,17 +1511,17 @@ int ObjectRef::l_hud_change(lua_State *L)
 	NO_MAP_LOCK_REQUIRED;
 	ObjectRef *ref = checkobject(L, 1);
 	RemotePlayer *player = getplayer(ref);
-	if (player == NULL)
+	if (player == nullptr)
 		return 0;
 
-	u32 id = lua_isnumber(L, 2) ? lua_tonumber(L, 2) : -1;
+	u32 id = luaL_checkint(L, 2);
```

Comment:
Assuming Krock's related comment is correct, the issue with `luaL_checkint` applies here too (strings are valid numbers, but the function won't accept them).

Reply:
(assuming he's right) why exactly someone should pass a string instead of a number though? The API says nothing about it

Reply:
This is called type coercion. Lua should convert strings to numbers when used in a numeric context - although, the documents are unclear as to whether this is just with arithmetic or with other contexts too

Reply:
I mean, that's useful for things like `print("this is a number: " .. 55)` but I still don't see why someone should pass a string into this function (`hud_change`). Anyways, a solution could be to have a `readParam<int>(...)` checking for NaN, applying type coercion and then casting into an int (if it's a decimal).  
Also, the previous version was wrong too, because I could have passed a float, and HUDs take no floats

Reply:
I don't know how important this really is (do other functions consistently take strings as numbers?), especially since you're not meant to do funny things to HUD IDs.